homeassistant/components/point/__init__.py

```diff
@@ -205,25 +234,23 @@ async def _sync(self):
 
         async def new_device(device_id, platform):
             """Load new device."""
-            config_entries_key = f"{platform}.{DOMAIN}"
-            async with self._hass.data[DATA_CONFIG_ENTRY_LOCK]:
-                if config_entries_key not in self._hass.data[CONFIG_ENTRY_IS_SETUP]:
-                    await self._hass.config_entries.async_forward_entry_setups(
-                        self._config_entry, [platform]
-                    )
-                    self._hass.data[CONFIG_ENTRY_IS_SETUP].add(config_entries_key)
-
             async_dispatcher_send(
                 self._hass, POINT_DISCOVERY_NEW.format(platform, DOMAIN), device_id
             )
 
         self._is_available = True
         for home_id in self._client.homes:
             if home_id not in self._known_homes:
+                await self._hass.config_entries.async_forward_entry_setups(
```

Comment:
We're only allowed to forward the config entry to a platform once. This looks like it could happen more than once for this config entry.

Reply:
What if more devices are added, what is the best approach for adding more devices on the fly?

Reply:
You already have a way to add new entities from new devices using the dispatch helper.